Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -146,14 +145,19 @@ class TopSitesPanel: UIViewController {
         if site is SuggestedSite {
             deleteTileForSuggestedSite(site as! SuggestedSite)
         }
-        
-        let newSites = profile.history.removeSiteFromTopSites(site) >>> {
-            self.profile.history.getTopSitesWithLimit(self.maxFrecencyLimit)
-        }
 
-        newSites.uponQueue(dispatch_get_main_queue()) { result in
-            self.deleteOrUpdateSites(result, indexPath: indexPath)
-            self.collection?.userInteractionEnabled = true
+        profile.history.removeSiteFromTopSites(site).uponQueue(dispatch_get_main_queue()) { _ in
+            // Remove the site from the current data source. Don't requery yet
```

Comment:
You should check `isSuccess` on the input here. If the removal failed, bail out.